memory/page_table.py

```diff
@@ -0,0 +1,75 @@
+#!/usr/bin/env python
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright: 2022 AMD 
+# Author: Kalpana Shetty <kalpana.shetty@amd.com>
+#
+
+import os
+from avocado import Test
+from avocado import skipUnless
+from avocado.utils import cpu, git, genio, build, process, linux_modules
+from avocado.utils.software_manager.manager import SoftwareManager
+
+
+class PageTable(Test):
+    """
+    Tests 5-level, 4-level page table depending on the supported platform.
+          * Test will detect CPU feature, 5 Level page table.
+          * Check for kernel support
+          * Run series of 5-level page tests from "pg-table_tests.git" that covers
+	        - heap, mmap, shmat tests.
+    :avocado: tags=memory
+    """
+
+    @skipUnless('x86_64' in cpu.get_arch(),
+                "This test runs on x86-64 platform.\
+        If 5-level page table supported on other platform then\
+        this condition can be removed")
+    def setUp(self):
+        smm = SoftwareManager()
+        for package in ['gcc', 'make']:
+            if not smm.check_installed(package) and not smm.install(package):
+                self.cancel('%s is needed for the test to be run' % package)
+
+        self.url = self.params.get(
+            'url', default="https://github.com/sanskriti-s/pg-table_tests.git")
```

Comment:
@kalpanashtty moving this to yaml will help more when we need to change the url.
So suggest to have this in yaml. ok to hard_coded in yaml if needed.

Reply:
Actually, I forgot to create yaml folder... will do.

Reply:
Done

Reply:
adding the hard_coded value for default url is not required here as we are already doing it in yaml. 
so please remove this and add `default=None`

Reply:
> adding the hard_coded value for default url is not required here as we are already doing it in yaml. so please remove this and add `default=None`

Done.